107/src/main/java/org/ehcache/jsr107/MultiCacheException.java

```diff
@@ -72,6 +73,27 @@ public String getMessage() {
     }
   }
 
+  MultiCacheException addFirstThrowable(Throwable t) {
+    if (t == null) {
+      throw new NullPointerException();
+    }
+
+    if (t == this) {
+      throw new IllegalArgumentException("cannot add to self");
+    }
+
+    if (t instanceof MultiCacheException) {
+      MultiCacheException multiCacheException = (MultiCacheException) t;
+      List<Throwable> throwables = multiCacheException.getThrowables();
```

Comment:
You're hiding the instance variable here... which means none of the code below does the right thing... right?